File: Assets/Scripts/LuaScripts/UI/UIGameOverPanel.lua
require("Core/System")

local UIManager = CS.Saro.UI.UIManager.Current

local M = class("UIGameOverPanel", target)

function M:OnAwake()
    self:Listen(self.btn_replay.onClick, M.OnClick_Replay)
    self:Listen(self.btn_setting.onClick, M.OnClick_Setting)
    self:Listen(self.btn_about.onClick, M.OnClick_About)
    self:Listen(self.btn_quit.onClick, CS.Saro.Main.Quit)
end

function M.OnClick_Replay()
    UIManager:AddToast("未开启")
end

function M.OnClick_Setting()
    UIManager:OpenUIAsync("UISetting")
end

function M.OnClick_About()
    UIManager:OpenUIAsync("UIAboutPanel")
end

return M

File: Assets/Scripts/UI/UIGameOverPanel.cs
using Cysharp.Threading.Tasks;
using Saro;
using Saro.UI;

namespace Tetris.UI
{
    public sealed partial class UIGameOverPanel : SingletonUI<UIGameOverPanel, UIBinder>
    {
        #region Impl

        protected override void InternalStart()
        {
        }

        protected override void InternalUpdate(float deltaTime)
        {
        }

        protected override void InternalClose()
        {
        }

        protected override void InternalAwake()
        {
            GetComps();

            Listen(btn_replay.onClick, OnClick_Replay);
            Listen(btn_setting.onClick, OnClick_Setting);
            Listen(btn_about.onClick, OnClick_About);
            Listen(btn_quit.onClick, () =>
            {
                Main.Quit();
            });
        }

        private void OnClick_Replay()
        {
            UIManager.Current.AddToast("未开启");
        }

        private void OnClick_Setting()
        {
            UIManager.Current.OpenUIAsync<UISetting>().Forget();
        }

        private void OnClick_About()
        {
            UIManager.Current.OpenUIAsync<UIAboutPanel>().Forget();
        }

        #endregion
    }

    // =============================================
    // code generate between >>begin and <<end
    // don't modify this scope

    //>>begin

    public partial class UIGameOverPanel
    {
        private UnityEngine.UI.Button btn_replay;
        private UnityEngine.UI.Button btn_setting;
        private UnityEngine.UI.Button btn_about;
        private UnityEngine.UI.Button btn_quit;

        private void GetComps()
        {
            btn_replay = Binder.Get<UnityEngine.UI.Button>("btn_replay");
            btn_setting = Binder.Get<UnityEngine.UI.Button>("btn_setting");
            btn_about = Binder.Get<UnityEngine.UI.Button>("btn_about");
            btn_quit = Binder.Get<UnityEngine.UI.Button>("btn_quit");
        }
    }

    //<<end

    // =============================================
}
